From 67d03a98c2cae5f41843c897f206adde299b0509 Mon Sep 17 00:00:00 2001
From: John Blackbourn <johnbillion@git.wordpress.org>
Date: Wed, 29 Nov 2017 16:00:48 +0000
Subject: [PATCH] Hardening: Remove the ability to upload JavaScript files for
 users who do not have the `unfiltered_html` capability.

Built from https://develop.svn.wordpress.org/trunk@42261


git-svn-id: http://core.svn.wordpress.org/trunk@42090 1a063a9b-81f0-0310-95a4-ce76da25c4cd
---
 wp-includes/functions.php | 5 +++--
 wp-includes/version.php   | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/wp-includes/functions.php b/wp-includes/functions.php
index 5a2d3c5d4eea..c5f6feb2f09e 100644
--- a/wp-includes/functions.php
+++ b/wp-includes/functions.php
@@ -2565,8 +2565,9 @@ function get_allowed_mime_types( $user = null ) {
 	if ( function_exists( 'current_user_can' ) )
 		$unfiltered = $user ? user_can( $user, 'unfiltered_html' ) : current_user_can( 'unfiltered_html' );
 
-	if ( empty( $unfiltered ) )
-		unset( $t['htm|html'] );
+	if ( empty( $unfiltered ) ) {
+		unset( $t['htm|html'], $t['js'] );
+	}
 
 	/**
 	 * Filters list of allowed mime types and file extensions.
diff --git a/wp-includes/version.php b/wp-includes/version.php
index cee0d424cda4..279f795fed1e 100644
--- a/wp-includes/version.php
+++ b/wp-includes/version.php
@@ -4,7 +4,7 @@
  *
  * @global string $wp_version
  */
-$wp_version = '5.0-alpha-42260';
+$wp_version = '5.0-alpha-42261';
 
 /**
  * Holds the WordPress DB revision, increments when changes are made to the WordPress DB schema.
